Initial commit for NAD D 7050 component

mwsluis opened the pull request:
Initial commit for NAD D 7050 media player component. For homeassistant.io documentation see doc-nad7050 pull request

Commit:
Initial commit for NAD D 7050 component

Initial commit for NAD D 7050 component

homeassistant commented:
Hi @mwsluis, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#6882).

Once you do that we will be able to review and accept this pull request.

Thanks!

mwsluis commented:
So this is my first contribution to home assistant and Git in general. For reviewers: I'm not so familiar with socket handling and catching all possible exceptions. Please have a look at the code in the send method. Also I have not been able run tox successfully on my machine.

Commit:
corrected lint errors

Commit:
fixed logging-not-lazy check

armills commented:
It's OK to use travis to run tox, especially for your first PRs. It can sometimes be tricky to set up.

All of the low-level protocol info ([example](https://github.com/home-assistant/home-assistant/pull/6882/files#diff-e1abd12400a125084826f724a68e4ea7R173)) will [need to be moved to an external library](https://home-assistant.io/developers/add_new_platform/#interfacing-with-devices). You might want to reach out to @joopert to see if this can be included in the existing [nad_receiver](https://github.com/joopert/nad_receiver) library. Supporting both transports via the same library would be the best solution.

mwsluis commented:
@armills Thanks for your input. I moved the low level stuff to the nad_receiver library as suggested, and added to my component as dependency.

mwsluis commented:
It seems there is another NAD product that uses tcp for remote control. I need to do some research to see if I can directly support it. I think I should therefor rename this component.

armills commented:
A more generic rename would be good, and even better if it's possible would be to integrate the TCP receiver protocol into the existing nad platform, so that users configure either a serial port or host/port. [Here's one example of such a platform](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/sensor/kwb.py). The hope would be that some of the extra functionality could be shared, [such as the volume helpers](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/media_player/nad.py#L84).

Also want to take a minute to say that this is really coming together well. Appreciate the work!

armills commented:
To elaborate, it would be nice if [here](https://github.com/joopert/nad_receiver/blob/master/nad_receiver/__init__.py), we could create a generic `NADReceiver` class that both the serial and TCP classes inherit from. This class can describe an abstract interface with methods such as `power_on`, `power_off`, etc. Then the serial class can implement this method as `self.main_power('=', 'On')`. Then in hass we can fetch either class based on how the user configures it, but can call the same methods for both.

pvizeli commented on homeassistant/components/media_player/nad7050.py:
```diff
@@ -0,0 +1,180 @@
+    @property
+    def source_list(self):
+        """List of available input sources."""
+        return self.d7050.available_sources()
```
Do that make IO?

mwsluis replied:
To the device? No.

cribbstechnologies committed:
repairing functionality for non-zero based ranges

cribbstechnologies committed:
fixing documentation

cribbstechnologies committed:
fixing nits

pezinek committed:
Forecasts for weather underground (#7062)

andrey-git committed:
Switch russound, pymysensors, and pocketcasts to pypi (#7449)

* Switch russound to pypi

* Switch pymysensors to pypi

* Switch pocketcasts to pypi

MartinHjelmare committed:
Upgrade pymysensors to 0.10.0 (#7469)

MartinHjelmare committed:
Upgrade pymysensors to 0.10.0 (#7469)

balloob committed:
Capitalize group names in demo

balloob committed:
Demo: Update old group member thermostat.ecobee -> climate

balloob committed:
Upgrade Dockerfile to Python 3.6 (#7471)

balloob committed:
Test only dependencies (#7472)

* Generate requirements file for tests

* Update tox

* Update validate

* Lint

* Tweak order in travis.yml to run longest job first

finish06 committed:
Update to pyunifi 2.12 (#7468)

* Update to pyunifi 2.12

* Update requirements_all.txt

finish06 committed:
Update to pyunifi 2.12 (#7468)

* Update to pyunifi 2.12

* Update requirements_all.txt

balloob committed:
Update requirements_test_all.txt

robbiet480 committed:
Make this an error instead of an info

balloob committed:
Remove listening to homeassistant_start with event automation (#7474)

frog32 committed:
Fix plant MIN_TEMPERATURE, MAX_TEMPERATURE validation (#7476)

* Fix plant MIN_TEMPERATURE, MAX_TEMPERATURE validation

small_float only allows values from 0 to 1 so we should use float instead

* Do not use vol.All for a single validation

frog32 committed:
Fix plant MIN_TEMPERATURE, MAX_TEMPERATURE validation (#7476)

* Fix plant MIN_TEMPERATURE, MAX_TEMPERATURE validation

small_float only allows values from 0 to 1 so we should use float instead

* Do not use vol.All for a single validation

balloob committed:
Update frontend

balloob committed:
Merge pull request #7482 from home-assistant/release-0-44-1

0.44.1

balloob committed:
Version bump to 0.44.1

balloob committed:
Update tox.ini

balloob committed:
Update tox.ini

balloob committed:
Fix travis

gurumitts committed:
Uses pypi for deps (#7485)

balloob committed:
0.44.2 (#7488)

* Version bump to 0.44.2

* Update frontend

amelchio committed:
LIFX: avoid out-of-bounds hue aborting the colorloop effect (#7495)

The hue is now a float but the hsbk conversion still believed it to be
an integer that could not be larger than 359. The float can in fact be,
for example, 359.9 and this would cause an out-of-bounds error in the
set_color call.

For completeness, the initial hue is also changed to a float.

fabaff committed:
Upgrade async_timeout to 1.2.1 (#7490)

aequitas committed:
Prevent printing of packets. (#7492)

A small bug in the python-rflink library caused packets to be printed. This update prevents this from happening.

fabaff committed:
Upgrade beautifulsoup4 to 4.6.0 (#7491)

andrey-git committed:
Switch onkyo to pypi (#7497)

Commit:
Fixed potential AttributeError when checking for deleted sources (#7502)

emlove committed:
Refactor sun component for correctness (#7295)

* Refactor sun component for correctness

* Convert datetimes to dates for astral

* Fix tests for updated code

* Fix times now that calcs are fixed

* Move sun functions to helpers

* Fix flake on new file

* Additional tweaks from review

* Update requirements

abmantis committed:
new source only forces "play" if the current state is "playing" (#7506)

andersonshatch committed:
Correct retrieval of spotify shuffle state (#7505)

Returned on the current playback response itself, not the device

frog32 committed:
Fix sonos sleep timer (#7503)

balloob committed:
Add support for automation config panel (#7509)

* Add support for automation config

* Build fromtend

* Lint

turbokongen committed:
Zwave panel api (#7456)

* # This is a combination of 3 commits.
# The first commit's message is:
Add seperate zwave panel

# The 2nd commit message will be skipped:

#	unused import

# The 3rd commit message will be skipped:

#	Use get for config

* Add seperate zwave panel

* more info

* Add usercodeview

* Improve api

* Improve api

* Separate api into own file.

* disable missing import

* review changes

* Tests 1

* Verify that we fetch data from groups

* Tests groups

* config 1

* usercode 1

* Api mods

* Tweak API

* docstrings

* 100% api testing

frog32 committed:
Update docker dev environment to python3.6 (#7520)

* Update docker dev environment to python3.6

* comment out disable switches again

andrey-git committed:
Switch basicmodem and python-roku to pypi (#7514)

imrehg committed:
sensor.envirophat: do not set up platform if hardware is not attached (#7438)

* sensor.envirophat: do not set up platform if hardware is not attached

Fixes comment from:
https://github.com/home-assistant/home-assistant/pull/7427#discussion_r114703904

* Fix update logic.

azogue committed:
Telegram Bot enhancements with callback queries and new notification services  (#7454)

* telegram_bot and notify.telegram enhancements:
- Receive callback queries and produce `telegram_callback` events.
- Custom reply_markup (keyboard or inline_keyboard) for every type of message (message, photo, location & document).
- `disable_notification`, `disable_web_page_preview`, `reply_to_message_id` and `parse_mode` optional keyword args.
- Line break between title and message fields: `'{}\n{}'.format(title, message)`
- Move Telegram notification services to `telegram_bot` component and forward service calls from the telegram notify service to the telegram component, so now the `notify.telegram` platform depends of `telegram_bot`, and there is no need for `api_key` in the notifier configuration. The notifier calls the new notification services of the bot component:
	- telegram_bot/send_message
	- telegram_bot/send_photo
	- telegram_bot/send_document
	- telegram_bot/send_location
	- telegram_bot/edit_message
	- telegram_bot/edit_caption
	- telegram_bot/edit_replymarkup
	- telegram_bot/answer_callback_query
- Added descriptions of the new notification services with a services.yaml file.
- CONFIG_SCHEMA instead of PLATFORM_SCHEMA for the `telegram_bot` component, so only one platform is allowed.
- Async component setup.

* telegram_bot and notify.telegram enhancements: change in requirements_all.txt.

nunofgs committed:
Add password parameter to uvc component (#7499)

fabaff committed:
Don't interact with hass directly (#7099)

basschipper committed:
Support for the PiFace Digital I/O module (#7494)

* Added rpi_pfio component supporting the PiFace I/O module

* Fixed some code style issues

* Removed global listener

* Update rpi_pfio.py

corneyl committed:
Upgrade limitlessled to 1.0.7 (#7525)

fabaff committed:
Update docstrings and log messages (#7526)

emlove committed:
Try to request current_location Automatic scope (#7447)

fabaff committed:
Add myStrom binary sensor (#7530)

fabaff committed:
Add not-context-manager (#7523)

* Add not-context-manager

* Add missing comma

Kane610 committed:
Threadsafe configurator (#7536)

* Make Configurator thread safe, get_instance timing issues breaking configurator working on multiple devices

* No blank lines allowed after function docstring

* Fix comment Tox

deftdawg committed:
Fix for #7459 (#7544)

* Generate a new updateDate with every call

This should fix #7459
Tests need to be updated in another commit.

* Replace STATIC_TIME with datetime object check

Removing the "DATE" argument from the Alexa component's configuration (because it is now dynamically generated) requires this commit's changes to the test cases to check that the updateDate data is a datetime type rather than a specific hardcoded value ('2016-10-10T19:51:42.0Z').

* Fix brackets

fabaff committed:
Upgrade sendgrid to 4.1.0 (#7538)

emlove committed:
Automatic version bump (#7555)

fabaff committed:
Upgrade dweepy to 0.3.0 (#7550)

tboyce021 committed:
Add SSL support to NZBGet sensor (#7553)

balloob committed:
Do not install all dependencies in dev mode (#7548)

* ps - do not install all dependencies

* Comment out blinkt because it depends on GPIO

* Add pip upgrade check back

* Disable import error blinkt

* Update comment

* Fix comment

emlove committed:
Hide proximity updates in logbook (#7549)

balloob committed:
Only install tox in dev mode (#7557)

tsvi committed:
Support adding different server locations for Microsoft face component (#7532)

* Support adding different server locations

* Rename variables and move CONF_ const into component as requested in review

* Fix unittests

* Forgot to add tests for microsoft_face_identify

andrey-git committed:
Treat swing and fan level as optional in Sensibo Climate. (#7560)

Kane610 committed:
Axis component (#7381)

* Added Axis hub, binary sensors and camera

* Added Axis logo to static images

* Added Axis logo to configurator
Added Axis mdns discovery

* Fixed flake8 and pylint comments

* Missed a change from list to function call
V5 of axis py

* Added dependencies to requirements_all.txt

* Clean up

* Added files to coveragerc

* Guide lines says to import function when needed, this makes Tox pass

* Removed storing hass in config until at the end where I send it to axisdevice

* Don't call update in the constructor

* Don't keep hass private

* Unnecessary lint ignore, following Baloobs suggestion of using NotImplementedError

* Axis package not in pypi yet

* Do not catch bare excepts. Device schema validations raise vol.Invalid.

* setup_device still adds hass object to the config, so the need to remove it prior to writing config file still remains

* Don't expect axis.conf contains correct values

* Improved configuration validation

* Trigger time better explains functionality than scan interval

* Forgot to remove this earlier

* Guideline says double qoutes for sentences

* Return false from discovery if config file contains bad data

* Keys in AXIS_DEVICES are serialnumber

* Ordered imports in alphabetical order

* Moved requirement to pypi

* Moved update callback that handles trigger time to axis binary sensor

* Renamed configurator instance to request_id since that is what it really is

* Removed unnecessary configurator steps

* Changed link in configurator to platform documentation

* Add not-context-manager (#7523)

* Add not-context-manager

* Add missing comma

* Threadsafe configurator (#7536)

* Make Configurator thread safe, get_instance timing issues breaking configurator working on multiple devices

* No blank lines allowed after function docstring

* Fix comment Tox

* Added Axis hub, binary sensors and camera

* Added Axis logo to static images

* Added Axis logo to configurator
Added Axis mdns discovery

* Fixed flake8 and pylint comments

* Missed a change from list to function call
V5 of axis py

* Added dependencies to requirements_all.txt

* Clean up

* Added files to coveragerc

* Guide lines says to import function when needed, this makes Tox pass

* Removed storing hass in config until at the end where I send it to axisdevice

* Don't call update in the constructor

* Don't keep hass private

* Unnecessary lint ignore, following Baloobs suggestion of using NotImplementedError

* Axis package not in pypi yet

* Do not catch bare excepts. Device schema validations raise vol.Invalid.

* setup_device still adds hass object to the config, so the need to remove it prior to writing config file still remains

* Don't expect axis.conf contains correct values

* Improved configuration validation

* Trigger time better explains functionality than scan interval

* Forgot to remove this earlier

* Guideline says double qoutes for sentences

* Return false from discovery if config file contains bad data

* Keys in AXIS_DEVICES are serialnumber

* Ordered imports in alphabetical order

* Moved requirement to pypi

* Moved update callback that handles trigger time to axis binary sensor

* Renamed configurator instance to request_id since that is what it really is

* Removed unnecessary configurator steps

* Changed link in configurator to platform documentation

* No blank lines allowed after function docstring

* No blank lines allowed after function docstring

* Changed discovery to use axis instead of axis_mdns

* Travis CI requested rerun of script/gen_requirements_all.py

amelchio committed:
LIFX: add lifx_set_state service call (#7552)

* Move service helpers to LifxManager

* Add lifx_set_color

This is a synonym for light.turn_on except it does not actually turn on the
light unless asked to.

Thus, turn_on can be implemented just by asking.

* Rename set_color to set_state

* Support power=False with lifx_set_state

florincosta committed:
Add raspihats binary sensor (#7508)

* Added raspihats binary_sensor platform

* Updated .coveragerc to ommit raspihats platforms.

* Using vol.Coerce(int) for validation and casting of I2CHat config address

perosb committed:
update pywebpush to 1.0.0 (#7561)

aequitas committed:
Initialize sun with correct values. (#7559)

* Initialize sun with unknown values.

Initial values should be `unknown` instead of `0`. Otherwise on HA restart the value of `0` is pushed to metrics databases (graphite/influx/recorder).

* Update sun position before emitting initial update

* Simplify based on armills comment.

* Use provided time for calculation.

bjw-s committed:
Comment RasPi specific requirements (#7562)

balloob committed:
Update netdisco (#7563)

gurumitts committed:
Support lutron serena shades (#7565)

* Adds support for the Lutron Caseta Serena shades hardware

* fixes typos

emlove committed:
Tests for zwave discovery logic (#7566)

* Tests for zwave discovery logic

* Simplify patching

* Test ignored node

emlove committed:
Tests for zwave setup features (#7570)

* Tests for zwave setup features

* Add test for frontend panel register

fronzbot committed:
Blink version bump (#7571)

Bumped blink version to support automatic reauthorization when tokens expire. Changed the battery sensor call to a string version so that the battery reports back "Low" or "OK" rather than a cryptic integer

bestlibre committed:
Fix systematic warning in influxdb sensor (#7541)

azogue committed:
Fix Kodi specific services registry and add descriptions (#7551)

* Fix Kodi specific services, add descriptions, add more handled exceptions

 - Fixes issue #7528
 - Add descriptions for Kodi specific services in services.yaml.
 - Error handling in Kodi API errors.
 - Make compatible the existent specific service `media_player.kodi_set_shuffle` with the general `media_player.shuffle_set` service (both use the same method but with different named parameter, I think the Kodi specific service should be eliminated, since it is not)

* fix line too long

* removed new services (for another PR); removed `kodi_set_shuffle` service

* requested changes

 - Removed `kodi_set_shuffle` service.
 - Optional `media_name` and `artist_name` parameters. `media_name` defaults to 'ALL'.
 - Guard clause to check if the services are already registered.

stu-gott committed:
Add Kira component to sensor and remote platforms (#7479)

* Add Kira component to sensor and remote platforms

* Test cases for Kira component and platforms

emlove committed:
Add tests for zwave network events (#7573)

Commit:
Additional Wink lock features (#7445)

* Additional Wink lock features

balloob committed:
Websocket_api: avoid parallel drain (#7576)

* Websocket_api: avoid parallel drain

* Remove send_message method

balloob committed:
Remove more test requirements (#7574)

* No longer require pyunify during tests

* No longer require cast during tests

* No longer required dependency for tests

* No longer require pymochad for tests

* Astral is a core dependency

* Avoid having to install datadog dependency during tests

* CMUS test doesn't test anything

* Frontier Silicon doesn't test anything

* No longer require mutagen

* Update requirements_test_all.txt

* Remove stale comment

mnoorenberghe committed:
camera.zoneminder: Handle old versions of zoneminder (#7589)

frog32 committed:
Pass additional arguments to tox in test_docker (#7591)

balloob committed:
Fix websocket api reaching queue (#7590)

* Fix websocket api reaching queue

* Fix outside task message sending

* Fix Py34 tests

azogue committed:
SMTP notify enhancements: full HTML emails and custom `product_name` in email headers (#7533)

* SMTP notify enhancements: HTML emails and customization options

- Send full HTML emails, with or without inline attached images.
- Custom `timeout`.
- Custom `product_name` identifier for the `FROM` and `X-MAILER` email headers.
- New HTML email test

* `sender_name` instead of product_name

 - Change `sender_name` instead of `product_name`.
 - No changes in `X-Mailer` header.
 - `From` header as always unless you define the new `sender_name` parameter.

emlove committed:
Automation State Change For timer attribute fix (#7584)

jhemzal committed:
Add posibility to specify snmp protocol version (#7564)

* snmp sensor protocol version configuration option

* fixed lint findings.

frog32 committed:
Add sonos alarm clock update service (#7521)

* Add sonos alarm clock update service

* Add tests and break lines

* Fix style errors

* Make test work with python<3.6

* Fix last two pylint errors

* fix new line to long errors

mezz64 committed:
Fix handling of single user (#7587)

fabaff committed:
File sensor (#7569)

* Add File sensor

* Use None and return

* Remove I/O

* Use less memory

* No traceback if file is empty

frog32 committed:
Make miflora monitored_conditions parameter optional (#7598)

* Make miflora monitored_conditions parameter optional.

* Use default keyword instead.

* Use list instead of tuple

* Simplify even more

balloob committed:
Force automation ids to always be a string (#7612)

balloob committed:
Update frontend

pschmitt committed:
Update Docker base image to python 3.6 (#7613)

olekenneth committed:
Add Content-type: image/jpeg for camera proxy (#7581)

* Add Content-type: image/jpeg for camera proxy

* Set content_type in constructur

fabaff committed:
Fix typo and update style to match the other platforms (#7621)

pavoni committed:
Bump pyvera - fixes issue with % in brightness levels. (#7622)

amelchio committed:
Add kelvin/brightness_pct alternatives to light.turn_on (#7596)

* Refactor color profiles to a class

* Refactor into preprocess_turn_on_alternatives

* LIFX: use light.preprocess_turn_on_alternatives

This avoids the color_name duplication and gains support for profile.

* Add kelvin parameter to light.turn_on

* Add brightness_pct parameter to light.turn_on

* LIFX: accept brightness_pct in effects

* Add test of kelvin/brightness_pct conversions

cnrd committed:
Add support for disabling tradfri groups (#7593)

* Add support for disabling tradfri groups

* Fixed styleguide problems

* Fix style problems

* Use default for groups when setting up in UI

fabaff committed:
Update docstrings and comments (#7626)

fabaff committed:
Upgrade Sphinx to 1.6.1 (#7624)

fabaff committed:
Update docstrings (#7630)

azogue committed:
Kodi specific service to call Kodi API methods (#7603)

* Kodi specific services to call Kodi API methods

 - new service: `kodi_execute_addon` to run a Kodi Addon with optional parameters. Results of the Kodi API call, if any, are redirected in a Home Assistant event: `kodi_execute_addon_result`.
 - new service: `kodi_run_method` to run a Kodi JSONRPC API method with optional parameters. Results of the Kodi API call are redirected in a Home Assistant event: `kodi_run_method_result`.
 - Add descriptions in services.yaml.
 - Add `timeout` parameter to yaml config (needed to make slow queries to the JSONRPC API, default timeout is set to 5s).
 - Trigger events with the results of the Kodi API calls, with:
 ```
 event_data = {
   'result': api_call_results,
   'result_ok': boolean,
   'input': api_call_parameters,
   'entity_id': 'media_player.kodi'}
```

* no need to clean OrderedDicts; no need for the `kodi_execute_addon` service

* no need for the `kodi_execute_addon` service

* unused import

* naming changes

corneyl committed:
Updated limitlessled requirement to v1.0.8 (#7629)

commento committed:
Osram lightify Removed wrong assignment (#7615)

self._brightness is assigned with the returned value of the
set_luminance() function, which is always equal to None.

danielperna84 committed:
Updated dependency (#7638)

fabaff committed:
Seven segments OCR image processing (#7632)

* Add initial seven segments OCR image processing

* Fix typo

balloob committed:
Abort tests when instances leaked (#7623)

amelchio committed:
Coerce color_temp to int even when passed in as kelvin (#7640)

balloob committed:
Fix automation failing to setup if no automations specified (#7647)

balloob committed:
Update frontend

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

balloob committed:
Version bump to 0.45

balloob committed:
Version bump to 0.46.0.dev0

fabaff committed:
Align with OpenALPR platform for naming conf variables (#7650)

fabaff committed:
Align with OpenALPR platform for naming conf variables (#7650)

tchellomello committed:
Added support to Amcrest camera to feed using RTSP via ffmpeg (#7646)

* Implemented ffmpeg option on Amcrest camera and upgraded to version 1.2.0

* Added ffmpeg arguments and binary options to Amcrest camera

* Added ffmpeg as dependencies

* Makes lint happy and fixed requirements_all.txt

* Inherent the ffmpeg.binary configuration from ffmpeg component

* Update amcrest.py

tchellomello committed:
Added support to Amcrest camera to feed using RTSP via ffmpeg (#7646)

* Implemented ffmpeg option on Amcrest camera and upgraded to version 1.2.0

* Added ffmpeg arguments and binary options to Amcrest camera

* Added ffmpeg as dependencies

* Makes lint happy and fixed requirements_all.txt

* Inherent the ffmpeg.binary configuration from ffmpeg component

* Update amcrest.py

happyleavesaoc committed:
bump fedex version (#7653)

happyleavesaoc committed:
bump ups version (#7654)

happyleavesaoc committed:
update usps (#7655)

* update usps

* fix doc

turbokongen committed:
Final tweaks for Zwave panel (#7652)

* # This is a combination of 3 commits.
# The first commit's message is:
Add seperate zwave panel

# The 2nd commit message will be skipped:

#	unused import

# The 3rd commit message will be skipped:

#	Use get for config

* Add seperate zwave panel

* Modify set_config_parameter to accept setting string values

* descriptions

* Tweaks

* Tweaks

* Tweaks

* Tweaks

* lint

* Fallback if no config parameteres are available

* Update services.yaml

* review changes

turbokongen committed:
Final tweaks for Zwave panel (#7652)

* # This is a combination of 3 commits.
# The first commit's message is:
Add seperate zwave panel

# The 2nd commit message will be skipped:

#	unused import

# The 3rd commit message will be skipped:

#	Use get for config

* Add seperate zwave panel

* Modify set_config_parameter to accept setting string values

* descriptions

* Tweaks

* Tweaks

* Tweaks

* Tweaks

* lint

* Fallback if no config parameteres are available

* Update services.yaml

* review changes

balloob committed:
Update frontend

balloob committed:
Update frontend

robbiet480 committed:
Add network_key as a config option (#7637)

* Add network_key as a config option

* Update __init__.py

robbiet480 committed:
Add network_key as a config option (#7637)

* Add network_key as a config option

* Update __init__.py

turbokongen committed:
Bugfix #7586 (#7661)

turbokongen committed:
Bugfix #7586 (#7661)

balloob committed:
Fix tests (#7659)

* Remove global hass

* Http.auth test no longer spin up server

* Remove server usage from http.ban test

* Remove setupModule from test device_sun_light_trigger

* Update common.py

fabaff committed:
Change line endings to LN (#7660)

thecynic committed:
Point pylutron to pypi (#7664)

balloob committed:
Update frontend

balloob committed:
Update frontend

Commit:
Upgrade Openhome library (#7671)

* Added support for openhome devices using transport service

* Style cleanup

amelchio committed:
Ignore attribute changes in automation trigger from/to (#7651)

* Ignore attribute changes in automation trigger from/to

* Quote names in deprecation warnings

This makes it somewhat easier to read if the suggestion happens to be
named "to".

* Add test with same state, new attribute value

andrey-git committed:
Switch pymodbus to pypi (#7677)

fabaff committed:
Merge pull request #7648 from home-assistant/release-0-45

0.45

fabaff committed:
Merge branch 'master' into dev

balloob committed:
Update state automation to work with new and deleted state changes

balloob committed:
Version bump to 0.45.1

tobygray committed:
zoneminder: fix incorrect use of logging.exception. (#7675)

Prior to this change the zoneminder component was attempting to
use logging.exception outside of exception handling code. This
would lead to the traceback module throwing an exception when
trying to work out the traceback for the exception.

This fixes the issue by changing the exception call into a
plain error logging call.

andrey-git committed:
Zwave: Apply refresh_node workaround on 1st instance only (#7579)

* Apply refresh_node workaround on 1st instance only

* Add another test

balloob committed:
Guard against new and removed state change events (#7687)

turbokongen committed:
ZWave: Add reset service to meters (#7676)

* Add reset service for command_class meters.

* Add reset service for command_class meters.

* cast index to const.py

emlove committed:
Remove defunct INSTALL_OPENZWAVE from Dockerfile (#7697)

azogue committed:
Fix telegram chats (#7689)

* bugfix for Telegram chat_ids

- Negative `chat_id`s for groups.
- Include `chat_id` in event data.
- Handle KeyError when receiving other types of messages, as `new_chat_member` ones, and send them as text.

* unused import

* fix double quote style, fix boolean expr, change warning msg

* mistake

* some more fixes

- fix if condition for msg bad fields.
- return True for a correct but not allowed or not recognized message: if not, the message arrives continuously.
- Allow to receive messages from unauthorized users if they come from authorized groups.

* support for `edited_message`s

- They come as normal messages, except for the 'edited_message' field instead of 'message'.

azogue committed:
Fix telegram chats (#7689)

* bugfix for Telegram chat_ids

- Negative `chat_id`s for groups.
- Include `chat_id` in event data.
- Handle KeyError when receiving other types of messages, as `new_chat_member` ones, and send them as text.

* unused import

* fix double quote style, fix boolean expr, change warning msg

* mistake

* some more fixes

- fix if condition for msg bad fields.
- return True for a correct but not allowed or not recognized message: if not, the message arrives continuously.
- Allow to receive messages from unauthorized users if they come from authorized groups.

* support for `edited_message`s

- They come as normal messages, except for the 'edited_message' field instead of 'message'.

cgtobi committed:
Fix playback control of web streams (#7683)

Web streams can't be paused and resumed later. That's why volumio stops them instead of pausing them.

cgtobi committed:
Fix playback control of web streams (#7683)

Web streams can't be paused and resumed later. That's why volumio stops them instead of pausing them.

amelchio committed:
Sort entities in default groups by name (#7681)

* Sort entities in default groups by name

* Cleanups from review

LvivEchoes committed:
Continue tracking device over dhcp lease table if wireless adapter not installed (#7690)

tobygray committed:
device_tracker.ubus: Handle empty results (#7673)

If OpenWRT isn't running the DHCP server then some OpenWRT hardware,
such as TP-Link TL-WDR3600 v1, can't determine the host
corresponding to an associated wifi client. This change handles that
by returning None when the request has no data in the result.

tobygray committed:
device_tracker.ubus: Handle empty results (#7673)

If OpenWRT isn't running the DHCP server then some OpenWRT hardware,
such as TP-Link TL-WDR3600 v1, can't determine the host
corresponding to an associated wifi client. This change handles that
by returning None when the request has no data in the result.

mezz64 committed:
Bump pyEight version (#7701)

fabaff committed:
Upgrade python-telegram-bot to 6.0.1 (#7704)

fabaff committed:
Make 'sender' as requirement for the config (fixes #7698) (#7706)

balloob committed:
Allow fetching hass.io panel without auth (#7714)

balloob committed:
Allow fetching hass.io panel without auth (#7714)

balloob committed:
Merge pull request #7686 from home-assistant/release-0-45-1

0.45.1

infamy committed:
update apcacccess used in apcupsd to 0.0.10, which fixes random file drop from apcaccess (#7722)

balloob committed:
Constrain requests to a version (#7725)

Add an optional extended description…

aronsky committed:
Asuswrt continuous ssh (#7728)

* Make ssh and telnet connections continuous in asuswrt

* Refactored SSH and Telnet connections into respective classes.

* Fixed several copy-paste typos and errors.

* More typos fixed.

* Small changes to arguments, to pass automated tests.

* Removed unsupported named arguments.

* Fixed a couple of mistakes in Telnet, and other lint errors.

* Added Telnet tests, and added lint exceptions.

* Removed comments from tests, as they irritated the hound.

bzillins committed:
Ensure https base_url in telegram bot (#7726)

antsar committed:
Prevent the random template filter from caching its output. Fixes #5678 (#7716)

nordeep committed:
Don't initialize mqtt components which have already been discovered (#7625)

* Don't initialize mqtt components which have already been discovered

* Fix string length

* Fix blank lines, fix constant name

* Remove globals. Remove JSON dump

* Add tests. Update grammar

* PEP8 style issue

* Add hyphen to object_id regex

* PEP8 style fix

amelchio committed:
LIFX: disable color features for white-only bulbs (#7742)

The product type is already established in order to decide the Kelvin range
so just reuse that information to disable color features for white-only lights.

Also change the breathe/pulse effects to be more useful for white-only
bulbs. For consistency, color bulbs set to a desaturated (i.e. white-ish)
color get the same default treatment as white-only bulbs.

antsar committed:
Don't block startup more than 60 seconds while waiting for components. (#7739)

Commit:
Always return rgb color of bulbs (#7743)

bjw-s committed:
Redesign monitored variables for hp_ilo sensor (#7534)

* Redesign monitored variables

Allow generating specific sensors without the need for template sensors

* Import 3rd party library inside update method

* Remove jsonpath_rw dependency

* Do not interfere with value_template or ilo_data output

Do not interfere with value_template or ilo_data output, this is now the responsibility of the user and should be handled in `configuration.yaml`

Fix UnusedImportStatement

Fix newline after function docstring

* Always output results to state

stu-gott committed:
Fix time_date sensor to update at predictable intervals (#7644)

* Fix time_date sensor to update at predictable intervals

* Delete automations.yaml

everix1992 committed:
Added new commands and functionality to the harmony remote component. (#7113)

* Added new commands and functionality to the harmony remote component.

-This includes the ability to optionally specify a number of times to repeat a specific command, such as pressing the volume button multiple times.
-Also added a new command that allows you to send multiple commands to the harmony at once, such as sending a set of channel numbers.
-Updated the unit tests for these changes.

* Fix flake8 coding violations

* Remove send_commands command and make send_command handle a single or list of commands

* Remove send_commands tests

* Update itach and kira remotes for new send_command structure. Fix pyharmony version in requirements_all.txt

* Fix incorrect variable name

* Fix a couple minor issues with remote tests

cribbstechnologies committed:
Template light (#7657)

* starting light template component

* linting/flaking

* starting unit tests from copypasta

* working on unit testing

* forgot to commit the test

* wrapped up unit testing

* adding remote back

* updates post running tox

* Revert "adding remote back"

This reverts commit 852c87ff9694dfc48e92b74fd9dbafbc164a2393.

* adding submodule back from origin

* updating submodule

* removing a line to commit

* re-adding line

* trying to update line endings

* trying to fix line endings

* trying a different approach

* making requested changes, need to fix tests

* flaking

* union rather than intersect; makes a big difference

* more tests passing, not sure why this one's failing

* got it working

* most of the requested changes

* hopefully done now

* sets; the more you know

balloob committed:
Update links.html

amigian74 committed:
Exclude filter for event types (#7627)

* add exclude filter for event types to recorder component

* corrected long line (279)

* change source code structure
add test for exclude event types

* code cleanup

* change source code structure

* Update __init__.py

* Update test_init.py

hoh committed:
Change setup script to use pip install instead of setup.py develop (#7756)

Using `python setup.py develop` did not manage to install the required dependencies.
This updates `script/setup` to use `pip install -e .` instead in order to resolve the required dependencies.

balloob committed:
Merge branch 'master' into dev

antsar committed:
Add icons to device tracker. (#7759)

emlove committed:
Rename of the zwave hass.data constants (#7768)

* Rename of the zwave hass.data constants

* Remove zwave since it is already implied

balloob committed:
Get rid of mock http component app (#7775)

* Remove mock_http_component from config tests

* Remove mock_http_component_app from emulated hue test

balloob committed:
Remove usage of event_loop fixture (#7776)

balloob committed:
assertEquals is deprecated (#7777)

emlove committed:
Support for GE Zwave fan controller (#7767)

* Support for GE Zwave fan controller

* Tests for zwave fan

* Add additional fan workarounds

balloob committed:
Replace executor with async_add_job (#7658)

* Remove executor

* Lint

* Lint

* Fix tests

azogue committed:
Fix Telegram Bot send file to multiple targets, snapshots of HA cameras, variable templating, digest auth (#7771)

* fix double template rendering when messages come from notify.telegram

* fix 'chat' information not present in callback queries

* better inline keyboards with yaml

To make a row of InlineKeyboardButtons you pass:
- a list of tuples like: `[(text_b1, data_callback_b1), (text_b2, data_callback_b2), ...]
- a string like: `/cmd1, /cmd2, /cmd3`
- or a string like: `text_b1:/cmd1, text_b2:/cmd2`

Example:
```yaml
data:
   message: 'TV is off'
   disable_notification: true
   inline_keyboard:
     - TV ON:/service_call switch.turn_on switch.tv, Other:/othercmd
     - /help, /init
```

* fix send file to multiple targets

* fix message templating, multiple file targets, HA cameras

- Allow templating for caption, url, file, longitude and latitude fields
- Fix send a file to multiple targets
- Load data with some retrying for HA cameras, which return 500 one or two times sometimes (generic cams, always!).
- Doc in services for new inline keyboards yaml syntax: `Text button:/command`

* HttpDigest authentication as proposed in #7396

* review changes

- Don't use `file` as variable name.
- For loop
- Simplify filter allowed `chat_id`s.

* Don't use `file` as variable name!

* make params outside the while loop

* fix chat_id validation when editing sent messages

balloob committed:
Fix more deprecation warnings (#7778)

* Remove setting up an hbmqtt broker

* Don't pass loop to web.Application in tests

* Use .query instead of deprecated .GET for aiohttp requests

* Fix closing file resource

* Do not use asyncio mark

* Notify.html5 - PyJWT: Use options to disable verify

* Yamaha: Test was still using deprecated ip

* Remove pytest-asyncio

sander76 committed:
Powerview to async (#7682)

* first commit

* first commit

* first commit

* first commit

* changing requirements

* updated requirements_all.txt

* various changes as suggested in the comments.

* using global values for dict keys.

CTLS committed:
Fix home/stay in concord232 (#7789)

scarface-4711 committed:
Add Marantz SSDP discovery / Detect error string in AppCommand.xml body (#7779)

fabaff committed:
Upgrade Sphinx to 1.6.2 (#7805)

fabaff committed:
Upgrade slacker to 0.9.50 (#7797)

Cinntax committed:
Bump pyenvisalink to version 2.1 (#7803)

fabaff committed:
Upgrade pysnmp to 4.3.6 (#7806)

Klikini committed:
Rachio (Sprinklers) (#7600)

* Rachio platform started

* Rachio tests

* detect bad api token

* Documentation, Code cleanup

* Docstrings end with a period, log uses %

* Fix arguments, default run time is now 10 minutes

* Fix typo, remove todo (GH issue exists)

* Revert polymer submodule commit

* Use a RachioPy version with SSL cert validation

* Update requirements

Teagan42 committed:
Rest notify data (#7757)

* Rest notify data

* Cleanup

* Fix spaces

aronsky committed:
Added handling of an AssertionError from pxssh failed login (#7750)

* Added handling of an AssertionError from pxssh failed login

* Destory and re-create pxssh instance, to fix behavior upon router restart.

fabaff committed:
Upgrade sqlalchemy to 1.1.10 (#7807)

fabaff committed:
Upgrade mypy to 0.511 (#7809)

Add an optional extended description…

aequitas committed:
Allow configuring DSMR5 protocol. (#7535)

* Allow configuring DSMR5 protocol.

* Give good example.

* Using dev branch until released upstream.

* Update to dsmr_parser supporting v5 arguments.

pvizeli commented:
Please fix travis and we can merge it :+1:

azogue committed:
Telegram send image: fix mimetype detection (#7802)

* Add `name` var to BytesIO content to get recognized

Sometimes the python-telegram-bot doesn't recognize the mimetype of the file and looks after a name variable to deduce it. Fixes #7413

* bytesio stream recycle less explicit

mezz64 committed:
Bump aiohttp to 2.1.0 (#7825)

pvizeli committed:
Cleanup telegram / Add url to webhook (#7824)

* Cleanup telegram / Add url to webhook

* fix lint

* Fix lint

fabaff committed:
Upgrade pysnmp to 4.3.7 (#7828)

Danielhiversen committed:
Flux led fix (#7829)

* Update flux_led.py

* style fix

happyleavesaoc committed:
update snapcast media player (#7079)

* update snapcast

* fix docstrings

* bump dep version

* address snapcast review comments

* add snapcast group volume support

* fix snapcast requirements

* update snapcast client entity id

* snapshot/restore functions

* refactor snapshot/restore services

* clean up

* update snapcast req

* bump version

* fix async updates

fabaff committed:
Update docstrings and log messages (#7709)

fabaff committed:
Upgrade PyMVGLive to 1.1.4 (#7832)

tchellomello committed:
Added persistent error message if cover.myq fails to load (#7700)

* Show persistent error if cover.myq fails

* Fixed typo on getLogger()

* Added ValueError on except condition

* Make pylint happy

* Removed DEFAULT_ENTITY_NAMESPACE since it is not being used

balloob committed:
Add initial benchmark framework (#7827)

* Add initial benchmark framework

* Use timer from timeit

philhawthorne committed:
Downgrade Docker to Python 3.5 to solve Segmentation Faults (#7799)

Downgrades the Dockerfiles used by Home Assistant to Python 3.5, after
Python 3.6 base image was causing segmentation faults.

See home-assistant/home-assistant#7752

tchellomello committed:
Introduced support to Netgear Arlo Cameras (#7826)

*  Introduced support to Netgear Arlo Cameras

* Using async_setup_platform() and applied other changes

* Removed unecessary variables

* Using asyncio for sensor/arlo

* Update arlo.py

* Removed entity_namespace

fabaff committed:
Allow 'base_url' (fixes #7784) (#7796)

emlove committed:
Expose the node name on the zwave node entity (#7787)

danielperna84 committed:
Bumped pyhomematic version (#7861)

emlove committed:
Rename time trigger 'after' to 'at' (#7846)

emlove committed:
Use expected behvaior for above/below (#7857)

azogue committed:
log time delay of domain setup in info level (#7808)

* log time delay of domain setup in info level

 * when setup problems appear, it's difficult to debug which are the components that took a lot to set up. This minimal change goes further than the 'slow setup warning' and measures the setup time interval for each domain.

* use timer as in helpers/entity

bokub committed:
Optimize history_stats efficiency and database usage (#7858)

abmantis committed:
dont use default for switch name, so that the object id is used (#7845)

amelchio committed:
Disallow ambiguous color descriptors in the light.turn_on schema (#7765)

* Disallow ambiguous color descriptors in the light.turn_on schema

* Update tests

balloob committed:
Make monkey patch work in Python 3.6 (#7848)

* Make monkey patch work in Python 3.6

* Update dockerfiles back to 3.6

* Lint

* Do not set env variable for dockerfile

* Lint

craigjmidwinter committed:
fix permissions issue for Insteon Local #6558 (#7860)

* fix unlinked commit

* Update insteon_local.py

infamy committed:
Start of migration framework, to allow moving of files in the config … (#7740)

* Start of migration framework, to allow moving of files in the config directory to be hidden, ios.conf used as the first one to undergo this change.

* Update const.py

* Update test_config.py

* improvement to syntax

Teagan42 committed:
Spotify aliases (#7702)

* Alias support for spotify devices

* Fix log

* Formatting/Fixes

* Remove default arg

* Add default keyword

* None check

titilambert committed:
Query in InfluxDB sensor is now templatable (#7634)

molobrakos committed:
fix for https://github.com/home-assistant/home-assistant/issues/7019 (#7618)

bjw-s committed:
Check if media commands are actually applicable (#7595)

* Check if media commands are actually applicable

- Explicitly allow ‘stop’ and ‘play’ on radio streams
- Disallow media commands when the playlist is empty
- Check if command is supported when calling `turn_on` and `turn_off`

* Suppress UPnP error 701 on media commands

* Clean up soco_filter_upnperror

Clean up soco_filter_upnperror and fix small bug in support_previous_track determination

DavidLP committed:
Quickfix Bug #7384 (#7582)

* Quickfix Bug #7384

* Fix devices not available runtime bug

aequitas committed:
Align switch group handling with light. (#7577)

balloob committed:
Update netdisco (#7865)

balloob committed:
Update frontend

balloob committed:
Version bump to 0.47.0dev0

molobrakos committed:
Update squeezebox.py (#7617)

Do not fail in case no players are connected, in which case squeezeserver will return a result without player_loop.

balloob committed:
Fix html5 unsub (#7874)

* Fix #7758 subscription expiration/removal

Removes a subscription after receiving an HTTP 410 response when trying to send a new message.

* Fix tests failing due to additional call

* Fix code style

* Lint

Mister-Espria committed:
Added effects to Yeelight bulbs (#7152)

* Added effects to Yeelight bulbs

* Fix Typo and Use randint instead of randrange

* Added Effects

* updated requirements_all.txt

* fix empty line

* minor fixes

* fix passing effects as parameter

twendt committed:
Enocean Binary Sensor: Handle click of both rockers (#7770)

emlove committed:
Support for renaming ZWave values (#7780)

* Support for renaming ZWave values

* Improve test

balloob committed:
Fix telegram_bot (#7877)

amelchio committed:
[light.lifx] Update aiolifx (#7882)

This makes LIFX Gen3 lights work with the current firmware.

joopert committed:
Change nad_receiver to pypi (#7852)

* Change to pypi

* add requirements

andrey-git committed:
Switch pymyq to pypi (#7884)

fabaff committed:
Use constants (#7888)

fabaff committed:
Upgrade pyasn1-modules to 0.0.9 (#7887)

fabaff committed:
Upgrade sendgrid to 4.2.0 (#7886)

fabaff committed:
Upgrade python-telegram-bot to 6.0.3 (#7885)

balloob committed:
Fix MQTT camera test (#7878)

Commit:
Added a Taps Aff binary sensor (#7880)

* Added a Taps Aff binary sensor

* PR Review updates

* Added a Taps Aff binary sensor

* PR Review updates

* Improved error handling

* Cosmetic changes (ordering, docstings, etc.)

Commit:
Fix for Travis errors

Commit:
Merge remote-tracking branch 'origin/nad-d-7050-component' into nad-d-7050-component

# Conflicts:
#	homeassistant/components/media_player/nad7050.py

mwsluis commented:
250+ commits? I think I borked up somewhere.

armills commented:
Easiest fix is to just copy out your files, create a new branch from dev, and paste them in.

Also, as mentioned above, if you can give it a more generic name like nad-ip or nad-tcp that would be good.

balloob commented:
https://xkcd.com/1597/ 😛

mwsluis commented:
Created PR #7955 to replace this one.

armills commented:
Closed for https://github.com/home-assistant/home-assistant/pull/7955
